pkg/daemon/update.go

```diff
@@ -1542,6 +1544,39 @@ func (dn *Daemon) writeUnits(units []ign3types.Unit) error {
 	return nil
 }
 
+func decodeContents(source, compression *string) ([]byte, error) {
+	var contentsBytes []byte
+
+	// To allow writing of "empty" files we'll allow source to be nil
+	if source != nil {
+		source, err := dataurl.DecodeString(*source)
+		if err != nil {
+			return []byte{}, err
```

Comment:
For each of these errors, consider providing a little more context by wrapping them, e.g.:

```golang
source, err := dataurl.DecodeString(*source)
if err != nil {
  return []byte{}, fmt.Errorf("could not decode file content string: %w", err)
}
```

This makes it easier to find where this error occurred if you're going through console logs.